src/GeniusYield/Types/TxMetadata.hs

```diff
@@ -0,0 +1,114 @@
+module GeniusYield.Types.TxMetadata (
+  GYTxMetadata (..),
+  GYTxMetadataValue,
+  gyTxMetaMap,
+  gyTxMetaList,
+  gyTxMetaNumber,
+  gyTxMetaBytes,
+  gyTxMetaText,
+  gyMetaTextChunks,
+  gyMetaBytesChunks,
+  mergeGYTransactionMetadata,
+  makeApiTransactionMetadata,
+  metadataFromApi,
+  metadataToApi,
+  metadataMsg,
+  metadataMsgs
+) where
+
+import qualified Cardano.Api as Api
+import qualified Data.Map as Map
+import           Data.ByteString     (ByteString)
+import           Data.Word           (Word64)
+import           GeniusYield.Imports (Text, coerce)
+
+
+
+newtype GYTxMetadata = GYTxMetadata (Api.TxMetadataInEra Api.BabbageEra)
+  deriving Show
+
+instance Semigroup GYTxMetadata where
+  (<>) mdx mdy = metadataFromApi $ metadataToApi mdx <> metadataToApi mdy
```

Comment:
Once the underlying type is `Api.TxMetadata`, can newtype derive this instance.

Reply:
Yes, of course.  Now it is working that way.